cpp/inc/bond/core/nullable.h

```diff
@@ -117,7 +118,7 @@ class nullable<T, Allocator, true>
 
     explicit
     nullable(const allocator_type& alloc)
-        : _alloc(alloc),
+        : allocator_type(alloc),
```

Comment:
I think this would be clearer if we used `Allocator` here, as that's the name of the base.